app/controllers/api/v2/hosts_controller.rb

```diff
@@ -112,6 +112,7 @@ def create
         @host = Host.new(host_attributes(host_params))
         @host.managed = true if (params[:host] && params[:host][:managed].nil?)
         apply_compute_profile(@host)
+        @host.suggest_default_pxe_loader if params[:host][:pxe_loader].nil?
```

Comment:
@lzap AFAICT, this actually should work for attributes inherited from the hostgroup, because the call to `host_attributes` above should inherit any attributes not set on the host from the hostgroup, therefor if you don't pass an os parameter but the hostgroup has one it will inherit and work correctly.